client/state/wp-api/posts/fetch-post.js

```diff
@@ -0,0 +1,15 @@
+import wpcom from 'lib/wp';
+
+import { receivePost } from 'state/posts/actions';
+
+const savePost = dispatch => post => dispatch( receivePost( post ) );
+const failPost = error => console.log( error );
+
+export const fetchPost = ( { dispatch }, { siteId, postId } ) => {
```

Comment:
I would prefer moving `fetchPost` and `requirePost` closer to `state/posts/actions.js`. It's a little difficult at the moment to see the relationship between `requirePost` and `fetchPost` without having to open a number of files.